app.py

```diff
@@ -52,5 +56,88 @@ def actualise_cpu_data():
     cpu_service.calculate_cpu_usage()
 
 
+## *******  HISTORICAL DATA STORAGE AND VIEWING *********
+@app.route('/api/cpu')
+def get_cpu_usage():
+
+    cpu_usage = psutil.cpu_percent()
+    return jsonify(cpu_usage=cpu_usage)
+
+@app.route('/api/memory')
+def get_memory_usage():
+
+    memory_usage = psutil.virtual_memory().percent
+    return jsonify(memory_usage=memory_usage)
+
+@app.route('/api/disk')
+def get_disk_usage():
+
+    disk_usage = psutil.disk_usage('/').percent
+    return jsonify(disk_usage=disk_usage)
+
+@app.route('/api/network')
+def get_network_io():
+
+    network_io = psutil.net_io_counters().bytes_sent + psutil.net_io_counters().bytes_recv
+    return jsonify(network_io=network_io)
+
+@app.route('/historic-data')
+def render_historic_data():
+    conn = sqlite3.connect('system_metrics.db')
+    c = conn.cursor()
+
+    c.execute("SELECT * FROM metrics")
+    data = c.fetchall()
+
+    conn.close()
+
+    return render_template('historic_data.html', data=data)
+
+def collect_metrics():
```

Comment:
Very good! Now you can put this into each service, instead of in the main entry point of this app 😊

Reply:
put it into each service, I'm sorry I am sure I understand what you mean.

Reply:
You could see that each resource had a single responsible in "services" folder. The idea is to separate each

Reply:
Wait, so I am supposed to have 11 .py files inside the services folder??